src/components/TodoModal/TodoModal.tsx

```diff
@@ -15,25 +40,39 @@ export const TodoModal: React.FC = () => {
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              {`Todo #${id}`}
             </div>
 
             {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button type="button" className="delete" data-cy="modal-close" />
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={() => {
+                setSelectedTodo('');
```

Comment:
Ensure that the `selectedTodo` state is correctly reset across all components when the modal is closed. This is important for maintaining the correct application state.